src/index.ts

```diff
@@ -23,19 +23,53 @@ const defaultApps = [
   require('./plugins/default')
 ]
 
-export const createProbot = (options: Options) => {
-  options.webhookPath = options.webhookPath || '/'
-  options.secret = options.secret || 'development'
+export class Probot {
+  public server: express.Application
+  public webhook: any
+  public logger: Logger
+
+  private options: Options
+  private robots: Robot[]
+  private app: () => string
+
+  constructor(options: Options) {
+    options.webhookPath = options.webhookPath || '/'
+    options.secret = options.secret || 'development'
+
+    this.options = options
+    this.logger = logger
+
+    this.webhook = new Webhooks({path: options.webhookPath, secret: options.secret})
+
+    this.app = createApp({
```

Comment:
Next up is refactoring `createApp` into a class 😉

Reply:
I have created an issue for this #496, I will start working on that as soon as this ongoing PR is merged